pkg/appgw/internaltypes.go

```diff
@@ -220,3 +224,33 @@ func formatHostname(hostName string) string {
 	// Hostname is NOT empty - prefix it with a dash
 	return fmt.Sprintf("%s-", hostName)
 }
+
+func (listenerID *listenerIdentifier) setHostNames(hostnames []string) int {
+	listenerID.HostName = ""
+	hostNameCount:= int(math.Min(float64(len(hostnames)), float64(MaxAllowedHostnames)))
+	for i := 0; i < hostNameCount; i++ {
+		if listenerID.HostName == "" {
+			listenerID.HostName = hostnames[i]
+		}
+		listenerID.HostNames[i] = hostnames[i]
+	}
+	return hostNameCount
+}
+
+// Returns the hostnames as a slice
+func (listenerID *listenerIdentifier) getHostNames() ([]string, error) {
+	var hostnames []string
```

Comment:
shouldn't `getHostNames` be as simple as `return listenerID.HostNames` ?

Reply:
listenerID.Hostnames is easily accessible if you have listenerID. This is mainly to get it as a slice instead of array which is the format used in sdks